### PR TITLE
 [Feature Request] : Adding Kadane's Algorithm for Maximum Subarray Sum in an Array |   isuue no.  #115 

### LINKED ISSUE #115: [Feature Request] : Adding Kadane's Algorithm for  Maximum Subarray Sum in an Array
### Describe the solution you'd like
i Want to to add Kadane's Algorithm under Array section under DSA ,
kadane's Algorithm is use to find  Maximum Subarray Sum in an Array and it is one of the first array algorithm that a dsa aspirant encounters.

---

**Would you like to work on this feature?**  
- 
YES

i want to contribute to you project under Gssocextd and hacktoberfest

### PR DESCRIPTION
**Pull request by Hamza1821**
### Description
 i have added Kadane's Algorithm
Fixes #115

### Type of change

- New feature (non-breaking change which adds functionality)
- 

### Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my code
- I have commented my code, particularly in hard-to-understand areas
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/basic-dsa/array/arrays-kadanesalgorithm-dsa.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+Kadane's algorithm is a **dynamic programming algorithm** used to find the maximum sum of a contiguous subarray within a one-dimensional array of numbers. It efficiently calculates the largest sum that can be obtained by adding consecutive elements in the array.
+
+### Characteristics:
+
+- **Dynamic Programming Approach**:
+  - Kadane's algorithm operates by maintaining a running sum of the maximum subarray ending at each position. It decides at every step whether to include the current element in the existing subarray or start a new subarray from the current element.
+
+- **Linear Time Algorithm**:
+  - The algorithm works in **O(n)** time complexity, where **n** is the number of elements in the array, as it only requires a single traversal of the array.
```
Comment (ajay-dhangar) [inline docs/basic-dsa/array/arrays-kadanesalgorithm-dsa.md]: replace `**O(n)**` with `$O(n)$` and replace `**n**` with `$n$`

File: docs/basic-dsa/array/arrays-kadanesalgorithm-dsa.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+- **Handles Negative Numbers**:
+  - Kadane’s algorithm can handle arrays with both positive and negative numbers, and it efficiently skips over negative subarrays that reduce the overall sum.
+
+### Time Complexity:
+
+- **Best, Average, and Worst Case: O(n)**
```
Comment (ajay-dhangar) [inline docs/basic-dsa/array/arrays-kadanesalgorithm-dsa.md]: replace `O(n)` with `$O(n)$`

File: docs/basic-dsa/array/arrays-kadanesalgorithm-dsa.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  - Kadane’s algorithm can handle arrays with both positive and negative numbers, and it efficiently skips over negative subarrays that reduce the overall sum.
+
+### Time Complexity:
+
+- **Best, Average, and Worst Case: O(n)**  
+  Kadane's algorithm processes each element exactly once, making its time complexity linear.
+
+### Space Complexity:
+
+- **Space Complexity: O(1)**
```
Comment (ajay-dhangar) [inline docs/basic-dsa/array/arrays-kadanesalgorithm-dsa.md]: replace `O(1)` with `$O(1)$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by Hamza1821** [comment]
please review.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**